src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py

```diff
@@ -1045,3 +1059,53 @@ def test_run_with_compute_instance_session_yml(
 
         run = pf.stream(run)
         assert run.status == RunStatus.COMPLETED
+
+    def test_session_id_with_different_env(self, pf: PFClient, randstr: Callable[[str], str]):
+        run = pf.run(
+            flow=f"{FLOWS_DIR}/flow_with_environment",
+            data=f"{DATAS_DIR}/env_var_names.jsonl",
+            name=randstr("name1"),
+        )
+        assert run.properties[RUNTIME_PROPERTY] == "automatic"
+        session_id_1 = run.properties[SESSION_ID_PROPERTY]
+
+        # same flow will get same session id
+        run = pf.run(
+            flow=f"{FLOWS_DIR}/flow_with_environment",
+            data=f"{DATAS_DIR}/env_var_names.jsonl",
+            name=randstr("name2"),
+        )
+        session_id_2 = run.properties[SESSION_ID_PROPERTY]
+        assert session_id_2 == session_id_1
+
+        with tempfile.TemporaryDirectory() as temp:
+            temp = Path(temp)
+            shutil.copytree(f"{FLOWS_DIR}/flow_with_environment", temp / "flow_with_environment")
+            # update image
+            flow_dict = load_yaml(temp / "flow_with_environment" / "flow.dag.yaml")
+            flow_dict["environment"]["image"] = "python:3.9-slim"
+
+            with open(temp / "flow_with_environment" / "flow.dag.yaml", "w", encoding="utf-8") as f:
+                dump_yaml(flow_dict, f)
+
+            run = pf.run(
+                flow=temp / "flow_with_environment",
+                data=f"{DATAS_DIR}/env_var_names.jsonl",
+                name=randstr("name3"),
+            )
+            session_id_3 = run.properties[SESSION_ID_PROPERTY]
+
+            assert session_id_3 != session_id_2
+
+            # update requirements
+            with open(temp / "flow_with_environment" / "requirements", "w", encoding="utf-8") as f:
```

Comment:
not requirements.txt but requirements?

Reply:
We do not restrict the file name in environment

Reply:
https://pip.pypa.io/en/stable/reference/requirements-file-format/